admin-dev/themes/new-theme/js/components/grid/extension/catalog/position-extension.js

```diff
@@ -0,0 +1,185 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import tableDnD from "tablednd/dist/jquery.tablednd.min";
+
+const $ = window.$;
+
+/**
+ * Class CategoryPositionExtension extends Grid with reorderable category positions
+ */
+export default class CategoryPositionExtension {
+    constructor() {
+        return {
+            extend: (grid) => this.extend(grid),
+        }
+    }
+
+    /**
+     * Extend grid
+     *
+     * @param {Grid} grid
+     */
+    extend(grid) {
+        this.grid = grid;
+        this._addIdsToGridTableRows();
+        grid.getContainer().find('.js-grid-table').tableDnD({
+            onDragClass: 'position-row-while-drag',
+            dragHandle: '.js-drag-handle',
+            onDragStart: () => {
+                this.originalPositions = decodeURIComponent($.tableDnD.serialize());
+            },
+            onDrop: (table, row) => this._handlePositionChange(row),
+        });
+    }
+
+    /**
+     * When position is changed handle update
+     *
+     * @param {HTMLElement} row
+     *
+     * @private
+     */
+    _handlePositionChange(row) {
+        const $rowPositionContainer = $(row).find('.js-' + this.grid.getId() + '-position:first');
+        const updateUrl = $rowPositionContainer.data('update-url');
+        const method = $rowPositionContainer.data('update-method');
+        const paginationOffset = parseInt($rowPositionContainer.data('pagination-offset'));
+        const positions = this._getRowsPositions(paginationOffset);
+        let params = {
+            updatedRowId: $rowPositionContainer.data('id'),
+            parentId: $rowPositionContainer.data('id-parent'),
+            positions: positions
+        };
+
+        this._updatePosition(updateUrl, params, method);
+    }
+
+    /**
+     * Returns the current table positions
+     * @returns {Array}
+     * @private
+     */
+    _getRowsPositions(paginationOffset) {
+        const tableData = JSON.parse($.tableDnD.jsonize());
+        const rowsData = tableData[this.grid.getId()+'_grid_table'];
+        const regex = /row_(\d+)_(\d+)/;
+
+        let positions = [];
+        for (let i = 0; i < rowsData.length; ++i) {
```

Comment:
The result of `.length` is not cached and recalculated on each loop. You should store the value in a variable.

Reply:
ok, is there the same issue below with `params.positions.length` ?